db/dohsql.c

```diff
@@ -1285,6 +1296,14 @@ int dohsql_end_distribute(struct sqlclntstate *clnt, struct reqlogger *logger)
         while (conns->conns[i].status != DOH_CLIENT_DONE) {
             Pthread_mutex_unlock(&conns->conns[i].mtx);
             poll(NULL, 0, 10);
+            if (bdb_lock_desired(thedb->bdb_env)) {
```

Comment:
NOTE: this should be safe imho since this is called in the parallel worker thread when calling write_response(RESPONSE_ROW).   We already call recover deadlock in this place for other reasons.